### PR TITLE
Rdm 6386 get case for case worker

### PR DESCRIPTION
**Pull request by MSancaktutar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6386

### Change description ###
@data-store-api: Implement functional test cases for "Get case for Case Worker" endpoint

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/aat/resources/features/F-040/F-040.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@F-040
+Feature: F-040: Get Case for Case worker
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-576
+  Scenario: must return successfully all and only the respective fields with READ access for Solicitors
+    Given a user with [an active Solicitor profile in CCD with a specific variation of ACLs on a case type]
+    And a user with [another active solicitor profile with different variation of ACLs on the same case type]
```
Comment (bmcts) [inline src/aat/resources/features/F-040/F-040.feature]: should we also mention the privileged user (solicitor 2) here?

File: src/aat/resources/features/F-040/F-040.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
+@F-040
+Feature: F-040: Get Case for Case worker
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-576
+  Scenario: must return successfully all and only the respective fields with READ access for Solicitors
+    Given a user with [an active Solicitor profile in CCD with a specific variation of ACLs on a case type]
```
Comment (bmcts) [inline src/aat/resources/features/F-040/F-040.feature]: does `a specific variation of ACLs` make sense? how about `Given a user with [an active solicitor profile in CCD with limited permissions on certain fields for a given case type]`

File: src/aat/resources/features/F-040/F-040.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    And a user with [another active solicitor profile with different variation of ACLs on the same case type]
+    And a successful call [to create a token for case creation] as in [Befta_Jurisdiction2_Default_Token_Creation_Data_For_Case_Creation]
+    And another successful call [by a privileged user with full ACL to create a case of this case type] as in [Befta_Jurisdiction2_Default_Full_Case_Creation_Data]
+    When a request is prepared with appropriate values
+    And the request [is prepared on behalf of the user creating the case above]
```
Comment (bmcts) [inline src/aat/resources/features/F-040/F-040.feature]: should mention that it is the privileged user making the call, eg. `And the request [is made by the privileged user who just created the case]`

File: src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+          "OPTION3"
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json]: Updated the code to use inheritance.

File: src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+          "classification": "PUBLIC",
+          "value": []
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json]: Updated the code to use inheritance.

File: src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    "body": {
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 0,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/aat/resources/features/common/case/befta-jurisdiction2-case-creation/Befta_Jurisdiction2_Default_Full_Case_Creation_Data.td.json]: Updated the code to use inheritance.

File: src/aat/resources/features/F-040/F-040.feature (hunk 1 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: It is actually Solicitor3 here. Updated the code to include solicitor 3 keyword.

File: src/aat/resources/features/F-040/F-040.feature (hunk 2 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: Updated the code to include "Given a user with [an active solicitor profile in CCD with limited permissions on certain fields for a given case type]".

File: src/aat/resources/features/F-040/F-040.feature (hunk 3 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: Agreed. Updated the code.

File: src/aat/resources/features/F-040/F-040.feature (hunk 1 above)
Comment (bmcts) [reply src/aat/resources/features/F-040/F-040.feature]: I mean in addition to solicitor 1 and solicitor 3, we should also mention solicitor 2 who is making calls in this scenario

File: src/aat/resources/features/F-040/F-040.feature (hunk 2 above)
Comment (bmcts) [reply src/aat/resources/features/F-040/F-040.feature]: can you also update the one below this to something similar?

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 0,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
+        "DocumentField4": {
+          "document_url": "http://dm-store:8080/documents/e56f0aea-4a5f-46d5-82d3-01b536420870",
```
Comment (bmcts) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json]: there is some url validation on the preview build which will reject this document_url. The domain needs to be changed from `dm-store:8080` to `dm-store-aat.service.core-compute-aat.internal` (so this url should be updated to `http://dm-store-aat.service.core-compute-aat.internal/documents/e56f0aea-4a5f-46d5-82d3-01b536420870`). Please can you update it here and all other occurrences in the PR

File: src/aat/resources/features/F-040/F-040.feature (hunk 1 above)
Comment (bmcts) [reply src/aat/resources/features/F-040/F-040.feature]: similar to this: https://github.com/hmcts/ccd-data-store-api/pull/738/files#diff-87a421b969e48e84889cb88e66334294R72-R74

Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: Updated the file @bmcts .

File: src/aat/resources/features/F-040/F-040.feature [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    And a user with [an active solicitor profile with another specific variation of ACLs on the same case type]
+    And a user with [an active Solicitor profile having full permissions on the same case type]
+    And another successful call [by a privileged user with full ACL to create a case of this case type] as in [Befta_Jurisdiction2_Default_Full_Case_Creation_Data]
+    When a request is prepared with appropriate values
+    And the request [is made by the privileged user who just created the case]
+    And it is submitted to call the [Get Case for Case Worker] operation of [CCD Data Store]
+    Then a positive response is received
+    And the response [contains values for all fields under the case type]
+    And the response has all other details as expected
+    And a call [to get the same case by Solicitor 1] will get the expected response as in [F-040_Later_Case_Read_By_Solicitor_1]
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040.feature]: Can you please make it clear that Solicitor 1 doesn't have READ permission on fields 1,3,6?

File: src/aat/resources/features/F-040/F-040.feature [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    And a call [to get the same case by Solicitor 3] will get the expected response as in [F-040_Later_Case_Read_By_Solicitor_3]
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040.feature]: Can you please make it clear that Solicitor 3 doesn't have READ permission on fields 2,5,7?

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
+        "DocumentField4": "PUBLIC",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+          "classification": "PUBLIC"
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
+        "DocumentField4": "PUBLIC",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 0,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
+        "DocumentField4": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+      "id": "[[ANY]]",
+      "jurisdiction": "BEFTA_JURISDICTION_2",
+      "state": "TODO",
+      "version": 0,
+      "case_type_id": "BEFTA_CASETYPE_2_1",
+      "created_date": "[[ANY]]",
+      "last_modified": "[[ANY]]",
+      "security_classification": "PUBLIC",
+      "case_data": {
+        "DocumentField4": {
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+        ],
+        "YesOrNoField": "Yes",
+        "EmailField": "testemail@gmail.com",
+        "TextField": "sometextfield",
+        "DateField": "2222-01-02",
+        "TextAreaField": "textarea",
+        "CollectionField": []
+      },
+      "data_classification": {
+        "DocumentField4": "PUBLIC",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+          "classification": "PUBLIC"
+        }
+      },
+      "after_submit_callback_response": null,
+      "callback_response_status_code": null,
+      "callback_response_status": null,
+      "delete_draft_response_status_code": null,
+      "delete_draft_response_status": null,
+      "security_classifications": {
+        "DocumentField4": "PUBLIC",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json]: Can use inheritance by creating a common file. Please check the feasibility.

File: src/aat/resources/features/F-040/F-040.feature (hunk 8 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: Updated the spec.

File: src/aat/resources/features/F-040/F-040.feature (hunk 9 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040.feature]: Updated the spec.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json (hunk 10 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_1.td.json]: Siva, it is being used at only one place. So we can skip the inheritance for now.

File: src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json (hunk 15 above)
Comment (nitishsingla1234) [reply src/aat/resources/features/F-040/F-040_Later_Case_Read_By_Solicitor_3.td.json]: Siva, it is being used at only one place. So we can skip the inheritance for now.

**Pull request merged**